### PR TITLE
Fixed out of scope variable error

### PR DESCRIPTION
**Pull request by Meg1456**

### CODE REVIEW

**Review by jiangyue12392 — COMMENTED** [review]
Hi Meg, I think there is a bit of indent issue in the else block and the way to fix out of scope error is probably to define `Message` variable outside instead of duplicating the `datastore` in both `if` and `else` branches

**Comment by Meg1456** [comment]
> Hi Meg, I think there is a bit of indent issue in the else block and the way to fix out of scope error is probably to define `Message` variable outside instead of duplicating the `datastore` in both `if` and `else` branches

Both the if and else blocks are using different strings to initiate the message variable. That is why I I had to duplicate it.. Is there another way to do this? 

Will fix the indentation :)

**Comment by jiangyue12392** [comment]
I think u can declare the Message outside and instantiates it separately in if and else block

**Comment by Meg1456** [comment]
Ok, fixed now. Is this better @jiangyue12392?

**Comment by jiangyue12392** [comment]
> Ok, fixed now. Is this better @jiangyue12392?

Yeah, sure

**Review by jiangyue12392 — APPROVED** [review]
Looks good to me

File: .factorypath [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<factorypath>
```
Comment (jiangyue12392) [inline .factorypath:1]: What is this file for though, if it is some local file path change then probably no need to push up

Comment (Reine0017) [reply .factorypath:1]: I'm also curious about this.

**Review by Reine0017 — COMMENTED** [review]
Hi @Meg1456, I think we should merge this commit first before the other files.

**Review by Reine0017 — CHANGES_REQUESTED** [review]
.factorypath

File: .factorypath (hunk 1 above)
Comment (potay) [reply .factorypath:1]: +1

**Review by potay — CHANGES_REQUESTED** [review]

File: .factorypath (hunk 1 above)
Comment (Meg1456) [reply .factorypath:1]: Im embarrassed to say even i have no clue what this file is. Didn't even realise it got push until you pointed it out... ill add it to .gitignore

Comment (Meg1456) [reply .factorypath:1]: Oh I think maybe when I switched IDEs, this file was auto generated for some reason..

**Comment by Meg1456** [comment]
Closing the PR. Creating a new one to fix the same issue - https://github.com/Reine0017/team-36-codeu/pull/26

**Pull request closed without merging**